zqd/handlers_zeek_test.go

```diff
@@ -16,98 +16,179 @@ import (
 	"github.com/brimsec/zq/zqd"
 	"github.com/brimsec/zq/zqd/api"
 	"github.com/brimsec/zq/zqd/packet"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"github.com/stretchr/testify/suite"
 )
 
-func TestPacketPost(t *testing.T) {
-	suite.Run(t, new(PacketPostSuite))
-}
-
-type PacketPostSuite struct {
-	suite.Suite
-	core     *zqd.Core
-	space    string
-	pcapfile string
-	payloads []interface{}
-}
+var (
+	zeekpath = os.Getenv("ZEEK")
+	cwd      = getwd()
+)
 
-func (s *PacketPostSuite) TestCount() {
-	expected := `
-#0:record[count:uint64]
-0:[4;]`
-	res := execSearch(s.T(), s.core, s.space, "count()")
-	s.Equal(test.Trim(expected), res)
+func TestPacketPostSuccess(t *testing.T) {
+	p := packetPost(t, zeekpath, "./testdata/valid.pcap", 202)
+	defer p.cleanup()
+	t.Run("DataReverseSorted", func(t *testing.T) {
+		expected := `
+#0:record[ts:time]
+0:[1501770880.988247;]
+0:[1501770877.501001;]
+0:[1501770877.471635;]
+0:[1501770877.471635;]`
+		res := execSearch(t, p.core, p.space, "cut ts")
+		assert.Equal(t, test.Trim(expected), res)
+	})
+	t.Run("SpaceInfo", func(t *testing.T) {
+		u := fmt.Sprintf("http://localhost:9867/space/%s", p.space)
+		var info api.SpaceInfo
+		httpJSONSuccess(t, zqd.NewHandler(p.core), "GET", u, nil, &info)
+		assert.Equal(t, p.space, info.Name)
+		assert.Equal(t, nano.Unix(1501770877, 471635000), *info.MinTime)
+		assert.Equal(t, nano.Unix(1501770880, 988247000), *info.MaxTime)
+		assert.InDelta(t, 1437, info.Size, 2)
+		assert.True(t, info.PacketSupport)
+		assert.Equal(t, p.pcapfile, info.PacketPath)
+		// XXX Cannot test with full struct because zeek's randomly generate
```

Comment:
The test verifies most of the fields in .info. Is there something specific that's not tested due to the changes between zeek runs? If so, could you slim this comment down & just state that?

Reply:
No, the test was just more readable when it test against the whole struct. I had to add the InDelta on `info.Size` to account for the change. Yeah I'll just comment on that.